File: RUST/CountCommasInRangeII.rs
/**
 * @version RUST 1.95.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ rustc --test CountCommasInRangeII.rs -o /tmp/runner && /tmp/runner
 */
pub struct CountCommasInRangeII;

impl CountCommasInRangeII {
    pub fn count_commas(n: i64) -> i64 {
        let mut ans = 0;
        let mut x = 1000;

        while x <= n {
            ans += n - x + 1;
            x *= 1000;
        }
        ans
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_count_commas() {
        // Teste 1
        let want = 3;
        let got = CountCommasInRangeII::count_commas(1002);
        assert_eq!(got, want);

        // Teste 2
        let want = 0;
        let got = CountCommasInRangeII::count_commas(998);
        assert_eq!(got, want);
    }
}

File: JAVA/CountCommasInRangeII.java
/**
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package JAVA;

public class CountCommasInRangeII {

    public static void main(String[] args) {
        System.out.printf("Resultado: %d\n", countCommas( 123 ));
    }

    public static long countCommas(long n) {
        long ans = 0;
        for (long x = 1000; x <= n; x *= 1000) {
            ans += n - x + 1;
        }
        return ans;
    }
}
